app/controllers/posts_controller.rb

```diff
@@ -34,34 +36,22 @@ def create
     end
     #debugger
     if @post.save
-      redirect_to root_path
+      redirect_to post_path(@post)
     else
-      redirect_to root_path, notice: @post.errors.full_messages.first
+      render "new", notice: @post.errors.full_messages.first
```

Comment:
So in flash messages you are showing errors. Explain. Why?

Reply:
I only know flash messages to show errors in case my validations do not work properly.

Reply:
How will the flash messge help the user to figure out what wrongs in his form. Flash message will be remove after few seconds. How will you communicate user after that?